### PR TITLE
creating new OpenSearch module

### PR DESCRIPTION
**Pull request by alexandru-rusanda**

### CODE REVIEW

File: versions.tf [hunk 1]
```diff
@@ -0,0 +1,12 @@
+terraform {
+  required_version = ">= 1.0"
+  required_providers {
+    aws = {
+      source = "hashicorp/aws"
+    }
+    elasticsearch = {
```
Comment (saritdevops) [inline versions.tf:7]: Is there any new provider we can maybe use or do we keep the old? @alexandru-rusanda

File: main.tf [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+      "product" = "opensearch",
+      "application_role" = "network",
+      "created_by" = "terraform"}
+    )
+  )
+
+  depends_on = [aws_iam_service_linked_role.os]
+}
+
+provider "elasticsearch" {
```
Comment (saritdevops) [inline main.tf:90]: why do we need this?

File: main.tf [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+          }
+        ]
+  ])
+}  
+
+########################################################################
+# This works only with export of AWS credentials not with sso login    #
+########################################################################
+
+resource "elasticsearch_opendistro_ism_policy" "cleanup" {
```
Comment (saritdevops) [inline main.tf]: I think we do not need ISM policies in the current implementation of opensearch.

File: main.tf [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+  }
+
+  advanced_options = var.advanced_options
+
+  access_policies = <<CONFIG
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Action": "es:*",
```
Comment (saritdevops) [inline main.tf:65]: We are allowing everyone full control over the ES? also to delete?

**Review by saritdevops — CHANGES_REQUESTED** [review]
I didnt find your readme. Can you point me to it ?
Please take a look at the comments

File: versions.tf (hunk 1 above)
Comment (alexandru-rusanda) [reply versions.tf:7]: I do not see any official one. This provider is the most popular.

File: main.tf (hunk 2 above)
Comment (alexandru-rusanda) [reply main.tf:90]: We used it to interact with the resources (URL) supported by Elasticsearch/Opensearch

File: main.tf (hunk 3 above)
Comment (alexandru-rusanda) [reply main.tf]: Ok...I'm gonna clean this up

File: main.tf (hunk 4 above)
Comment (alexandru-rusanda) [reply main.tf:65]: What is to be expected here ? Policy is created basd on the old elasticsearch frame

File: variables.tf [hunk 5]
```diff
@@ -0,0 +1,15 @@
+variable "env_name" {}
+variable "env_type" {}
+variable "security_group_ids" {}
+variable "os_group" {}
+variable "create_os" {}
+variable "vpc_private_management_subnet_id" {}
+variable "create_service_link_role" {}
+# variable "access_policie" {}
```
Comment (saritdevops) [inline variables.tf]: remove commented code.

**Review by saritdevops — APPROVED** [review]

File: variables.tf (hunk 5 above)
Comment (alexandru-rusanda) [reply variables.tf]: done...removed commented code also from main

File: main.tf [hunk 6]
```diff
@@ -9,14 +9,7 @@ resource "aws_opensearch_domain" "os" {
   for_each              = var.create_os ? var.os_group : {}
   domain_name           = "${each.value.domain_name}-${var.env_name}"
   engine_version        = each.value.engine_version
-
-#  # advanced_options = var.advanced_options
-#domain_endpoint_options {
```
Comment (saritdevops) [inline main.tf]: I think we do need certificate and custom endpoint actually (could be for version2) 
something like this:
custom_endpoint = "${each.value.domain_name}.${replace(var.env_name,"-","")}.tolunainsights-internal.com"

**Review by saritdevops — CHANGES_REQUESTED** [review]

**Review by saritdevops — APPROVED** [review]

**Pull request merged**